src/ngraph/runtime/executable.cpp

```diff
@@ -81,6 +81,7 @@ void runtime::Executable::validate(const vector<std::shared_ptr<runtime::Tensor>
     for (size_t i = 0; i < results.size(); i++)
     {
         if (outputs[i]->get_element_type().is_static() &&
+            results[i]->get_element_type() != element::dynamic &&
```

Comment:
@diyessi would we want this to be results[i]->get_element_type().is_static() as well? Looks like we are treating 'undefined' type as a static type as well?